shop/src/main/java/ro/msg/learning/shop/Repositories/CustomerRepository.java

```diff
@@ -0,0 +1,14 @@
+package ro.msg.learning.shop.Repositories;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.transaction.annotation.Transactional;
+import ro.msg.learning.shop.Entities.Customer;
+
+import java.util.Optional;
+
+@Transactional(readOnly = true)
+@EnableJpaRepositories(basePackages = "ro.msg.learning.shop.Repositories")
```

Comment:
if the repositories package is not a sub-package of the Spring main class package we need to declare the repositories packages, as shown here:@EnableJpaRepositories(basePackages = "ro.msg.learning.shop.Repositories")
This has to be provided in a configuration class or SpringBootApplication class.

Reply:
or use @Repository and Spring will automatically pick them up during the component scanning process.